pyron-core/src/test/scala/com/cloudentity/pyron/rule/PreparedRewriteTest.scala

```diff
@@ -0,0 +1,164 @@
+package com.cloudentity.pyron.rule
+
+import org.junit.runner.RunWith
+import org.scalatest.{MustMatchers, WordSpec}
+import org.scalatestplus.junit.JUnitRunner
+import PreparedRewrite._
+import com.cloudentity.pyron.domain.flow.PathParams
+
+import scala.util.Try
+
+@RunWith(classOf[JUnitRunner])
+class PreparedRewriteTest extends WordSpec with MustMatchers {
+
+  "PreparedRewrite" should {
+    "fail to create prepared rewrite when pattern is not a valid regex pattern" in {
+      Try(PreparedRewrite("/pattern/with(unbalanced(parens)/should/{param}/fail", "/api", "/resources/{param}/{1}")).isFailure mustBe true
```

Comment:
PreparedRewrite goes through all the rules only when the pyron reads the conf, and keeping previous approach, it will immediately throw and die on invalid configuration, here specifically invalid regex specified by the pattern. Once conf is read, all PreparedRewrites are literally prepared :) guaranteed to be correct and can either match or not-match, but won't "fail". I think it's good approach and it was used so before for reading the config. Let me know if this explanation is sufficient.

Reply:
switched to using method 'prepare' instead of 'apply', which returns Try explicitly and further logic deals wit Try